docs/limitrange.md

```diff
@@ -0,0 +1,104 @@
+---
+linkTitle: "LimitRange"
+weight: 300
+---
+-->
+
+# `LimitRange` support in Pipeline
+
+## `LimitRange`s, `Requests` and `Limits`
+
+Taken from the [LimitRange in kubernetes docs](https://kubernetes.io/docs/concepts/policy/limit-range/).
+
+By default, containers run with unbounded [compute resources](/docs/concepts/configuration/manage-resources-containers/) on a Kubernetes cluster.
+With resource quotas, cluster administrators can restrict resource consumption and creation on a `namespace` basis.
+Within a namespace, a Pod or Container can consume as much CPU and memory as defined by the namespace's resource quota. There is a concern that one Pod or Container could monopolize all available resources. A LimitRange is a policy to constrain resource allocations (to Pods or Containers) in a namespace.
+
+A _LimitRange_ provides constraints that can:
+
+- Enforce minimum and maximum compute resources usage per Pod or Container in a namespace.
+- Enforce minimum and maximum storage request per PersistentVolumeClaim in a namespace.
+- Enforce a ratio between request and limit for a resource in a namespace.
+- Set default request/limit for compute resources in a namespace and automatically inject them to Containers at runtime.
+
+`LimitRange` are validating and mutating `Requests` and `Limits`. Let's look, *in a nutshell*, on how those work in Kubernetes.
+
+- **Requests** are not enforced. If the node has more resource available than the request, the container can use it.
+- **Limits** on the other hand, are a hard stop. A container going over the limit, will be killed.
+
+Resource types for both are:
+-   CPU
+-   Memory
+-   Ephemeral storage
+
+The next question is : how pods with resource and limits are run/scheduled ?
+The scheduler *computes* the amount of CPA and memory requests (using **Requests**) and tries to find a node to schedule it.
+
+Requests and limits can be applied to both Containers and Init Containers.
+- For init containers, the max of each type is taken
+- For containers, it sums all requests/limits for each containers
+
+This means, if you got the following:
+-   initContainer1 : 1 CPU, 100m memory
+-   initContainer2 : 2 CPU, 200m memory
+-   container1 : 1 CPU, 50m memory
+-   container2 : 2 CPU, 250m memory
+-   container3 : 3 CPU, 500m memory
+
+The computation will be:
+-   CPU : 2 (max init containers) + 6 (sum of containers) = 8 CPU
+-   Memory: 200m (max init containers) + 800m (sum of containers) = 1000m (1G)
+
+## Tekton support
+
+The way Limits and Requests works in Kubernetes is because it is assumed that all containers run in parallel, and init container run before, each one after the others.
+
+That assumption — containers running in parallel — is not true in Tekton. They do all start together (because there is no way around this) **but** the *entrypoint hack* is making sure they actually run in sequence and thus there is always only one container that is actually consuming some resource at the same time.
```

Comment:
i wonder if we could link to some docs on the entrypoint hack - i'm surprised our developer docs dont have much on this but maybe we could link to https://github.com/tektoncd/pipeline/tree/main/cmd/entrypoint#entrypoint ?

Reply:
Indeed, I think it would be worth enhancing the docs around what the `entrypoint` does 👼🏼

Reply:
💯